packages/sdk/src/Actions/EventAction.ts

```diff
@@ -526,21 +531,87 @@ export class EventAction extends DeployableTarget<
-   * @param {?TxParams & { chainId?: number }} [params] - Additional parameters for validation, including known events, logs, and chain ID.
+   * @param {TxParams & { transactionHash?: Hex, chainId?: number }} params - Additional parameters for validation, including, hash, known events, logs, and chain ID.
    * @returns {Promise<boolean>}
    */
   public async isActionStepValid(
     actionStep: ActionStep,
-    params?: TxParams & { chainId?: number },
+    params: TxParams & { hash?: Hex; chainId?: number },
   ) {
     if (actionStep.signatureType === SignatureType.EVENT) {
-      return await this.isActionEventValid(actionStep, params);
+      const eventParams = params as ValidateEventStepParams;
+      const signature = actionStep.signature;
+      let event: AbiEvent;
+      // Lookup ABI based on event signature
+      if (eventParams.knownEvents) {
+        event = eventParams.knownEvents?.[signature] as AbiEvent;
+      } else {
+        event = (events.abi as Record<Hex, AbiEvent>)[signature] as AbiEvent;
+      }
+
+      if (!event) {
+        throw new Error(`No known ABI for given event signature: ${signature}`);
+      }
+
+      if (this.isArraylikeIndexed(actionStep, event)) {
+        // If the field is indexed, we can't filter on it
+        throw new UnparseableAbiParamError(
+          actionStep.actionParameter.fieldIndex,
+          event,
+        );
+      }
+
+      eventParams.event = event;
+
+      // Use the provided logs, no need to fetch receipt
+      if (eventParams.logs) {
+        return this.isActionEventValid(actionStep, {
+          ...eventParams,
+        });
+      }
+
+      if (!params?.hash) {
+        throw new ValidationHashMissingError();
+      }
+      if (!params?.chainId) {
+        throw new ValidationChainIdMissingError();
+      }
+
+      const client = this._config.getClient({
+        chainId: params.chainId,
+      }) as PublicClient;
+      const receipt = await client.getTransactionReceipt({
+        hash: params.hash,
+      });
+      const decodedLogs = receipt.logs.map((log) => {
+        const { eventName, args } = decodeEventLog({
+          abi: [event],
+          data: log.data,
+          topics: log.topics,
+        });
+
+        return { ...log, eventName, args };
+      });
+
+      return this.isActionEventValid(actionStep, {
+        logs: decodedLogs,
+        ...params,
+      });
     }
     if (actionStep.signatureType === SignatureType.FUNC) {
-      return await this.isActionFunctionValid(
-        actionStep,
-        params as ValidateFunctionStepParams,
-      );
+      if (!params?.hash) {
+        throw new ValidationHashMissingError();
+      }
+      if (!params?.chainId) {
+        throw new ValidationChainIdMissingError();
+      }
+      const client = this._config.getClient({
+        chainId: params.chainId,
+      }) as PublicClient;
+      const transaction = await client.getTransaction({
```

Comment:
shouldn't we be spreading `params` into here? otherwise there's no real reason for `ValidateFunctionStepParams` to extend `GetTransactionParameters`

Reply:
Ended up keeping `params.hash` since I really liked simplifying the `TxParams`